src/route/dataset.ts

```diff
@@ -219,37 +233,43 @@ router.get('/', async (req: Request, res: Response) => {
 // Returns a shallow dto of the dataset with the given ID
 // Shallow gives the revisions and dimensions of the dataset only
 router.get('/:dataset_id', async (req: Request, res: Response) => {
-    const datasetID: string = req.params.dataset_id;
+    const datasetID: string = req.params.dataset_id.toLowerCase();
     const dataset = await validateDataset(datasetID, res);
     if (!dataset) return;
     const dto = await DatasetDTO.fromDatasetComplete(dataset);
     res.json(dto);
 });
 
+// DELETE /api/dataset/:dataset_id
+// Returns a shallow dto of the dataset with the given ID
+// Shallow gives the revisions and dimensions of the dataset only
+router.delete('/:dataset_id', async (req: Request, res: Response) => {
+    const datasetID: string = req.params.dataset_id.toLowerCase();
+    const dataset = await validateDataset(datasetID, res);
+    if (!dataset) return;
```

Comment:
Do we want a 404 here, or is it handled by a fall-through route?

Reply:
So no dataset means we've already set the status code on the response in the `validateDataset` function.  So no dataset we just return because we've done everything we need to... If that makes sense?

Reply:
ah ok I had missed that validateDataset was manipulating the response.

In that case, what about making it a middleware? Something like:

```typescript
const validateDataset = async (req, res, next) => {
  try {
    res.locals.dataset = await Dataset.findOneByOrFail({ id: req.params.dataset_id });
  catch(err) {
    res.status(404);
    return res.json({ message: 'Dataset not found.' });
  }
};
```

and then apply it to the routes that need the dataset:

```typescript
router.delete('/:dataset_id', validateDataset, async (req...) => {
  const dataset = res.locals.dataset as Dataset;
});
```

Reply:
Let look at maybe doing this as follow up and get this landed.  I am thinking about generally moving towards `findOneByOrFail` for most database calls.